urllib3/util/retry.py

```diff
@@ -169,10 +184,10 @@ def get_backoff_time(self):
 
         :rtype: float
         """
-        if self._observed_errors <= 1:
+        if len(self.history) <= 1:
             return 0
 
-        backoff_value = self.backoff_factor * (2 ** (self._observed_errors - 1))
+        backoff_value = self.backoff_factor * (2 ** (len(self.history) - 1))
```

Comment:
Also, do we want to reset the backoff every time we get a non-error response? So only look at the last sequence of contiguous errors?